client/client.go

```diff
@@ -131,14 +134,35 @@ func (a *ApiClient) Request(code AuthToken) *ApiRequest {
 type ApiRequest struct {
 	api *ApiClient
 	// authorization code to use for requests
-	auth AuthToken
+	auth   AuthToken
+	revoke ClientReauthorizeFunc
 }
 
 func (ar *ApiRequest) Do(req *http.Request) (*http.Response, error) {
 	if req.Header.Get("Authorization") == "" {
 		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ar.auth))
 	}
-	return ar.api.Do(req)
+	r, err := ar.api.Do(req)
+	if r != nil && r.StatusCode == http.StatusUnauthorized {
+		// invalid JWT-token; most likely expired an expired token
+		log.Info("Device unauthorized; attempting reauthorization")
```

Comment:
`expired an expired token` does not sound good. Maybe you can add something `Most likely the token expired and we will try to refresh it and then retry sending the original request.`

Reply:
oops..